xbmc/imagefiles/ImageLoader.cpp

```diff
@@ -0,0 +1,105 @@
+/*
+ *  Copyright (C) 2023 Team Kodi
+ *  This file is part of Kodi - https://kodi.tv
+ *
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *  See LICENSES/README.md for more information.
+ */
+
+#include "ImageLoader.h"
+
+#include "ServiceBroker.h"
+#include "TextureCache.h"
+#include "URL.h"
+#include "guilib/Texture.h"
+#include "pictures/Picture.h"
+#include "utils/StringUtils.h"
+#include "utils/log.h"
+
+#include <chrono>
+
+using namespace IMAGE_FILES;
+
+std::unique_ptr<CTexture> CImageLoader::GetTexture(
```

Comment:
This is most of the logical parts of `CImageLoader::DoWork()` in PR #23009. If that and this ends up acceptable solutions, then it can call this. I just want to reuse it for `ResizeTexture` for the HTTPImageTransformationHandler.